jit/jit-rules.c

```diff
@@ -530,6 +527,10 @@ int _jit_create_call_setup_insns
 		*struct_return = 0;
 		return_ptr = 0;
 	}
+	if(is_nested)
+	{
+		need_outgoing_word(&passing);
+	}
```

Comment:
Could you please clarify why all this stuff is re-ordered here?

Reply:
Hmm this looks like a mistake, as I didn't change the order in `_jit_create_entry_insns`. The ABI says the struct return pointer is the first argument, so the argument order should probably be `fn(return_ptr, frame_ptr)`, but this needs to be changed in `_jit_create_entry_insns` too.

I'll check it out tomorrow.